packages/apps/human-app/server/src/modules/governance/governance.service.ts

```diff
@@ -34,48 +47,37 @@ export class GovernanceService {
       provider,
     );
 
-    const currentBlock = await provider.getBlockNumber();
-    const lastScannedBlockKey = this.generateCacheKey('last-scanned-block');
-    const proposalListKey = this.generateCacheKey('proposal', 'list');
-
-    const cachedLastScannedBlock =
-      (await this.cacheManager.get<number>(lastScannedBlockKey)) ?? 0;
+    const currentBlock = await provider.getBlock('latest');
+    if (!currentBlock) {
+      this.logger.error('No latest block from RPC');
+      throw new Error('Blockchain node unavailable');
+    }
 
     const fromBlock =
-      cachedLastScannedBlock > 0
-        ? cachedLastScannedBlock + 1
-        : Math.max(0, currentBlock - N_BLOCKS_LOOKBACK);
+      (cachedLastScannedBlock?.number ?? 0) > 0
+        ? (cachedLastScannedBlock?.number ?? 0) + 1
+        : currentBlock.number - N_BLOCKS_LOOKBACK;
 
-    let proposalList: Proposal[] = [];
+    let allProposals: ProposalResponse[] = [];
     try {
       const newProposals = await this.getProposalCreatedEvents(
         contract,
         fromBlock,
-        currentBlock,
+        currentBlock.number ?? 0,
```

Comment:
```suggestion
        currentBlock.number,
```
Shouldn't number always be here if block is retrieved above?